### PR TITLE
Adding extrude to polygonmesh

### PR DESCRIPTION
**Pull request by jkoehoorn**
Adding a optional depth parameter to get a extruded polygonmesh.
Support holes in the polygonmesh.

### CODE REVIEW

**Comment by jkoehoorn** [comment]
capture

File: Babylon/Mesh/babylon.polygonmesh.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
         }
     }
 
+    function nearlyEqual(a: number, b: number, epsilon: number = 0.0001): boolean {
+        if (a === b) {
+            return true;
+        }
+        return Math.abs(a - b) < epsilon;
+    }
+
```
Comment (deltakosh) [inline Babylon/Mesh/babylon.polygonmesh.ts]: Why not using Tools.WithinEpsilon(point.x, result[0].x, 0.00001) ?

**Comment by RaananW** [comment]
I actually just wanted to ask - why not use (or add, as it doesn't exist) Vector2.equalsWithEpsilon ? they are both Vector2...

File: Babylon/Mesh/babylon.polygonmesh.ts [hunk 2]
```diff
@@ -152,13 +165,116 @@
...
+            result.setVerticesData(positions, VertexBuffer.PositionKind, updatable);
+            result.setVerticesData(normals, VertexBuffer.NormalKind, updatable);
+            result.setVerticesData(uvs, VertexBuffer.UVKind, updatable);
             result.setIndices(indices);
 
             return result;
-        }
+        } 
+
+       private addside(positions: any[], normals: any[], uvs: any[], indices:any[],bounds: any, points: PolygonPoints, depth:number, flip:boolean ){
```
Comment (deltakosh) [inline Babylon/Mesh/babylon.polygonmesh.ts]: Could it be addSide ?

**Comment by RaananW** [comment]
@jkoehoorn - thanks for the update. 
What is the difference between your nearlyEqual function and WithinEpsilon in Tools?

**Comment by jkoehoorn** [comment]
That function is not made by me, i just added code to add sides. So i dont know.

**Comment by deltakosh** [comment]
OK I'll fix it, it will be faster :)

**Comment by RaananW** [comment]
:-) Was just about to do that.

**Comment by jkoehoorn** [comment]
Thanks

 

Van: David Catuhe [mailto:notifications@github.com] 
Verzonden: dinsdag 26 mei 2015 20:05
Aan: BabylonJS/Babylon.js
CC: jkoehoorn
Onderwerp: Re: [Babylon.js] Adding extrude to polygonmesh (#522)

 

OK I'll fix it, it will be faster :)

**Pull request merged**